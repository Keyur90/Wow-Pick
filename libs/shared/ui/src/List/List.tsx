import React, { ReactNode } from 'react';
import { Container } from '../Container';
import { makeStyles } from '@material-ui/core/styles';

interface ListPropTypes {
  children: NonNullable<ReactNode>;
}

const useStyles = makeStyles((theme) => ({
  root: {
    borderBottom: '1px solid #E0E0E0',
    marginBottom: theme.spacing(2),
    backgroundColor: theme.palette.common.white,
  },
  list: {
    listStyle: 'none',
    margin: 0,
    padding: theme.spacing(2, 0),
    '& li': {
      fontSize: '1.25rem',
      fontWeight: 400,
      lineHeight: '1.875rem',
      color: theme.palette.common.black,
      padding: theme.spacing(0, 0, 1),
      '&:last-child': {
        borderBottom: 'none',
        padding: theme.spacing(0),
      },
      display: 'flex',
      alignItems: 'center',
      '& > span': {
        marginRight: theme.spacing(2),
      },
      '& > svg': {
        color: '#3A474E',
        marginRight: theme.spacing(1.5),
      },
    },
  },
}));

const List: React.FC<ListPropTypes> = React.memo(({ children }) => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Container>
        <ul className={classes.list}>{children}</ul>
      </Container>
    </div>
  );
});

export { List };
