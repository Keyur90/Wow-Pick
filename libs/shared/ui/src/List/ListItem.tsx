import React, { ReactNode } from 'react';

interface ListItemPropTypes {
  children: NonNullable<ReactNode>;
}

const ListItem: React.FC<ListItemPropTypes> = ({ children }) => <li>{children}</li>;

export { ListItem };
