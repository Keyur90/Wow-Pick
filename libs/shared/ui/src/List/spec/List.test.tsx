import React from 'react';
import { render } from '@testing-library/react';
import { List } from '../List';
import { ListItem } from '../ListItem';

test('renders Container', () => {
  const { container } = render(
    <List>
      <ListItem>List item 1</ListItem>
      <ListItem>List item 2</ListItem>
      <ListItem>List item 3</ListItem>
    </List>
  );

  expect(container).toMatchSnapshot();
});
