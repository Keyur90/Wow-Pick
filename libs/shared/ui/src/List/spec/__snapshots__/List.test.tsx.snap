// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Container 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiContainer-root makeStyles-root-3 MuiContainer-maxWidthLg"
    >
      <ul
        class="makeStyles-list-2"
      >
        <li>
          List item 1
        </li>
        <li>
          List item 2
        </li>
        <li>
          List item 3
        </li>
      </ul>
    </div>
  </div>
</div>
`;
